src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -89,8 +86,18 @@ export async function makeGoConfig(config: SamLaunchRequestArgs): Promise<GoDebu
         throw Error('missing launch.json, template.yaml, and failed to discover project root')
     }
 
-    let localRoot: string | undefined
-    let remoteRoot: string | undefined
+    // If the target is 'code' we need to adjust codeRoot to the location of the source
+    if (config.invokeTarget.target === 'code') {
+        const parts = config.handlerName.split(':')
+
+        if (parts.length !== 2) {
+            throw Error(`malformed Go handler name: ${config.handlerName}`)
```

Comment:
Let's give hints everywhere we possibly can, that explain the _expected_ format. (Probably we need to do this for all runtimes, out of scope for this PR ofc.)
```suggestion
            throw Error(`Go handler is not in expected "directory:handler" format: ${config.handlerName}`)
```